miden-crypto/src/merkle/smt/full/large/subtree/mod.rs

```diff
@@ -0,0 +1,212 @@
+        bitmask[bit_offset / 8] |= 1 << (bit_offset % 8);
+    }
+
+    #[inline]
+    fn get_bit(bitmask: &[u8], bit_offset: usize) -> bool {
+        (bitmask[bit_offset / 8] >> (bit_offset % 8)) & 1 != 0
+    }
+
+    pub fn from_vec(root_index: NodeIndex, data: &[u8]) -> Result<Self, DeserializationError> {
+        if data.len() < Self::BITMASK_SIZE {
+            return Err(DeserializationError::InvalidValue("Subtree data too short".into()));
+        }
+        let (bitmask, hash_data) = data.split_at(Self::BITMASK_SIZE);
+        let present_hashes: usize = bitmask.iter().map(|&byte| byte.count_ones() as usize).sum();
+        if hash_data.len() != present_hashes * Self::HASH_SIZE {
+            return Err(DeserializationError::InvalidValue("Invalid hash data length".into()));
+        }
+
+        let mut nodes = Map::new();
+        let mut hash_chunks = hash_data.chunks_exact(Self::HASH_SIZE);
+
+        // Process each potential node position
+        for local_index in 0..Self::MAX_NODES {
+            let bit_offset = (local_index as usize) * Self::BITS_PER_NODE;
+            let has_left = Self::get_bit(bitmask, bit_offset);
+            let has_right = Self::get_bit(bitmask, bit_offset + 1);
+
+            if has_left || has_right {
+                // Calculate depth for empty hash lookup
+                let node_depth_in_subtree = Self::local_index_to_depth(local_index);
+                let child_depth = root_index.depth() + node_depth_in_subtree + 1;
+                let empty_hash = *EmptySubtreeRoots::entry(SMT_DEPTH, child_depth);
+
+                // Get left child hash
+                let left_hash = if has_left {
+                    let hash_bytes = hash_chunks.next().ok_or_else(|| {
+                        DeserializationError::InvalidValue("Missing left hash data".into())
+                    })?;
+                    Word::try_from(hash_bytes).map_err(|_| {
+                        DeserializationError::InvalidValue("Invalid left hash format".into())
+                    })?
+                } else {
+                    empty_hash
+                };
+
+                // Get right child hash
+                let right_hash = if has_right {
+                    let hash_bytes = hash_chunks.next().ok_or_else(|| {
+                        DeserializationError::InvalidValue("Missing right hash data".into())
+                    })?;
+                    Word::try_from(hash_bytes).map_err(|_| {
+                        DeserializationError::InvalidValue("Invalid right hash format".into())
+                    })?
+                } else {
+                    empty_hash
+                };
+
+                let inner_node = InnerNode { left: left_hash, right: right_hash };
+                nodes.insert(local_index, inner_node);
+            }
+        }
+
+        // Ensure all hash data was consumed
+        if hash_chunks.next().is_some() {
+            return Err(DeserializationError::InvalidValue(
```

Comment:
nit: it feels odd to have the amount of `"string".into()` as argument. Are we bound to `DeserializationError` from `winter-utils`?